app/admin/models/Categories_model.php

```diff
@@ -105,4 +106,80 @@ public function scopeIsEnabled($query)
     {
         return $query->where('status', 1);
     }
+    
+    public function scopeWhereHasLocation($query, $locationId)
+    {
+        $query->whereHas('locations', function ($q) use ($locationId) {
+            $q->where('locations.location_id', $locationId);
+        });
+    }
```

Comment:
This method already exists within the Locationable trait which is used on Categories model so you can remove it here.

Reply:
You missed this